### ISSUE TITLE
Exception of jsonb type in PostgreSql

### ISSUE THREAD

**Issue by arsen1991** [etlbox/etlbox#86]
We are getting the exception below while we trying to compare columns with jsonb type in PostgreSql:

System.AggregateException: One or more errors occurred. (42804: column "ResponsibleGaming" is of type jsonb but expression is of type text)

**Comment by andreaslennartz** [comment]
Hi,

Thanks for reporting this bug. Could you give me some more details how to reproduce this error? E.g. which component(s) are you using, which data type and when this error occurs?

Thanks for efforts!

**Comment by arsen1991** [comment]
Hi!

We are getting this exception when the column has type 'jsonb' in PostgreSql.

**Comment by andreaslennartz** [comment]
Hi,

Sorry for not being clear enough.

Could you either provide me a code example that creates the error, or could you tell me:

- you are saying that you are comparing something. Not quite sure what that means? how does your data flow look like? Is the error related to a DbDestination?
- if yes, how does your data object look like? Specifically, which CLR type does "ResponsibleGaming" have? Is it a string, a byte array or some other object type?

Thanks for your input,

Andreas

**Comment by arsen1991** [comment]
Hi!

- We are using DbMerge to merge the data between 2 tables.
- ResponsibleGaming has object type in C# which mapped to ResponsibleGaming column with jsonb type in PostgreSql.

**Comment by andreaslennartz** [comment]
Hi again,

could you give some more details about the exact object type of ResponsibleGaming in your POCO?
E.g.
you will have something like:
```
public class MyDataType : MergeableRow {
   [UpdateColumn]
   public *object* ResponsibleGaming {get;set;}
}
... 
var merge = new DbMerge<MyDataRow>(..);
```` `

I am no quite sure if I fully understood what the property ResponsibleGaming`s data type is. 
If it is of type object, I assume you use something different to store data in it? Which type is that?
Or is it already some other strongly-type user-defined object used as type? 

Did you try to use a string value instead (so to basically convert and store your json in a property of type string, which is used to write into the jsonb postgres column)?

**Comment by arsen1991** [comment]
Hi!

We reproduced this case on simple tables. For example, we are trying to sync table1 between 2 databases where
CREATE TABLE public.table1 (
    id int4 NULL,
    col1 int4 NULL,
    col2 int4 NULL,
    colj jsonb NULL
);


The code is below:

var sourceCon = new PostgresConnectionManager("Server=localhost;Database=demo;User Id=postgres;password=postgres");
            var destCon = new PostgresConnectionManager("Server=localhost;Database=demodest;User Id=postgres;password=postgres");
            DbSource source = new DbSource(sourceCon, "table1");
            DbMerge multicast = new DbMerge(destCon, "table1") { MergeMode = ETLBox.DataFlow.MergeMode.Full, UseTruncateMethod = false };

            multicast.MergeProperties.IdColumns = new List<ETLBox.DataFlow.IdColumn> { new ETLBox.DataFlow.IdColumn() { IdPropertyName = "id" } };
            multicast.MergeProperties.CompareColumns = new List<ETLBox.DataFlow.CompareColumn> { new ETLBox.DataFlow.CompareColumn() { ComparePropertyName = "colj" } };
            multicast.MergeProperties.UpdateColumns = new List<ETLBox.DataFlow.UpdateColumn> { new ETLBox.DataFlow.UpdateColumn() { UpdatePropertyName = "colj" } };
            source.LinkTo(multicast);

            source.Execute();
            multicast.Wait();

**Comment by andreaslennartz** [comment]
Thanks! This example clarifies the bug. 
I will look into this and keep you updated on the progress.

**Comment by andreaslennartz** [comment]
Hi @arsen1991 ,
a quick update on this: I was able to reproduce and fix the bug. I will release the fix with next version, which I plan to release probably in 1 or 2 weeks. I'll add a comment when I released the new version.

**Comment by arsen1991** [comment]
Hi @andreaslennartz !
Thanks for support . We'll be waiting for the update .

**Comment by andreaslennartz** [comment]
Hi @arsen1991 ,

I just released version 2.3.1 (along with Postgres connection manager 1.3.1)
This should solve your issue - jsonb data types should now be fully supported. 
Let me know if you run into other problems.
